### DISCUSSION TITLE
Transcribing audios just from my microphone

### DISCUSSION THREAD

**Discussion by filipe-campos** [JamesBrill/react-speech-recognition#117] [Q&A]
The library is transcribing the audios from my microphone and also from other people if I'm on a call. I would like to just transcribe the audio from my microphone.

Does anyone have any idea if this is possible?

**Comment by JamesBrill** [comment]
Hi @filipe-campos The Web Speech API will only transcribe audio from your microphone (I'm assuming your microphone was picking up other voices from your computer speakers). The ability to distinguish between speakers depends on the implementation of the speech-to-text API on the backend (on Chrome, this is Google's speech-to-text cloud service) - `react-speech-recognition` just passes on the transcript it gets from that API to your React app. Multiple voice identification appears to be a relatively new concept in these services ([some docs by Google](https://cloud.google.com/speech-to-text/docs/multiple-voices)) - unfortunately, I don't think the native Chrome implementation or either of the [polyfills](https://github.com/JamesBrill/react-speech-recognition/blob/master/docs/POLYFILLS.md) used by this library currently support it.

The Web Speech API specification does not include any fields for multiple voices - even if a cloud service did support it, we would need to add something custom to this library to support something off-spec like that.

So I don't think there's a good solution for this yet, at least not for web apps. If this is a critical requirement for your application, you may need to do the following:
* Identify a speech-to-text service that supports multiple voice identification. Google [appears](https://cloud.google.com/speech-to-text/docs/multiple-voices) to offer this
* Create a backend endpoint that proxies audio streams from a browser client to Google's API
* Have that endpoint stream a list of transcripts back to the browser, one for each speaker
* Write a custom browser client for passing these transcripts to your React app

It's quite a lot of work! I have a few things on my TODO list that could support the above one day:
* An example Node API for streaming audio to Google and returning the results back
* A [polyfill](https://github.com/JamesBrill/react-speech-recognition/blob/master/docs/POLYFILLS.md) for the Web Speech API that uses this custom Node API. Such a polyfill could then be plugged into `react-speech-recognition`
* Support in `react-speech-recognition` for custom fields emitted by a polyfill (each one has its own unique features that don't fit the Web Speech API spec)
